### PR TITLE
crimson/common/tri_mutex: update the class comment

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/crimson/osd/object_context.h [hunk 1]
```diff
@@ -130,11 +130,11 @@ class ObjectContext : public Blocker,
   void put_lock_type(RWState::State type) {
     switch (type) {
     case RWState::RWWRITE:
-      return rwlock.unlock_for_write();
+      return lock.unlock_for_write();
```
Comment (tchaikov) [inline src/crimson/osd/object_context.h:133]: note, the `get` and `put` usage is difficult to reason and debug. we might need to use RAII or a safer interface to expose this API. for instance, see https://github.com/scylladb/seastar/blob/master/include/seastar/core/rwlock.hh .

Comment (tchaikov) [reply src/crimson/osd/object_context.h:133]: @rzarzynski this is a follow-up of our discussion last night.

Comment (tchaikov) [reply src/crimson/osd/object_context.h:133]: https://trello.com/c/0PyCIhAL

**Review by athanatos — APPROVED** [review]

**Comment by tchaikov** [comment]
changelog

- s/rwlock/lock/. i missed one of the "rwlock"s in my previous change.

**Pull request merged**